# Lista de perguntas
- Crianças e adolescentes fora da escola vs gasto governamental com educação
- População analfabeta vs gasto governamental com educação
- Taxa de crianças que completam a educação primária vs gasto governamental com educação

In [3]:
import os
dir(os)

['DirEntry',
 'F_OK',
 'GenericAlias',
 'Mapping',
 'MutableMapping',
 'O_APPEND',
 'O_BINARY',
 'O_CREAT',
 'O_EXCL',
 'O_NOINHERIT',
 'O_RANDOM',
 'O_RDONLY',
 'O_RDWR',
 'O_SEQUENTIAL',
 'O_SHORT_LIVED',
 'O_TEMPORARY',
 'O_TEXT',
 'O_TRUNC',
 'O_WRONLY',
 'P_DETACH',
 'P_NOWAIT',
 'P_NOWAITO',
 'P_OVERLAY',
 'P_WAIT',
 'PathLike',
 'R_OK',
 'SEEK_CUR',
 'SEEK_END',
 'SEEK_SET',
 'TMP_MAX',
 'W_OK',
 'X_OK',
 '_AddedDllDirectory',
 '_Environ',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__spec__',
 '_check_methods',
 '_execvpe',
 '_exists',
 '_exit',
 '_fspath',
 '_get_exports_list',
 '_walk',
 '_wrap_close',
 'abc',
 'abort',
 'access',
 'add_dll_directory',
 'altsep',
 'chdir',
 'chmod',
 'close',
 'closerange',
 'cpu_count',
 'curdir',
 'defpath',
 'device_encoding',
 'devnull',
 'dup',
 'dup2',
 'environ',
 'error',
 'execl',
 'execle',
 'execlp',
 'execlpe',
 'execv',
 'execve',
 'execvp',
 'execvpe',
 'exts

In [4]:
data_list = [file for file in os.listdir() if file.endswith(".csv")]
data_list

['Adolescents_out_of_school.csv',
 'Children_out_of_school.csv',
 'Government_exp_const_usd.csv',
 'Government_exp_gdp.csv',
 'Illetarate_perc_female.csv',
 'Illetarate_population.csv',
 'Mean_years_schooling.csv',
 'Survival_rate_primary_ed.csv',
 'Teacher_comp_perc_public_inst.csv']

In [5]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df_adol = pd.read_csv(data_list[0])
df_adol[df_adol["Flags"]=="National Estimation"]

NATMON_IND                                          Indicator LOCATION  \
29    OFST_2_F_CP  Out-of-school adolescents of lower secondary s...      URY   
30    OFST_2_F_CP  Out-of-school adolescents of lower secondary s...      URY   
31    OFST_2_F_CP  Out-of-school adolescents of lower secondary s...      URY   
45    OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      CHE   
46    OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      CHE   
...           ...                                                ...      ...   
2234  OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      ISR   
2235  OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      ISR   
2236  OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      ISR   
2237  OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      ISR   
2238  OFST_2_M_CP  Out-of-school adolescents of lower secondary s...      ISR   

          Country  TIME  Time   Value Flag Codes                Flags  
29        Uruguay  2017  2017  1171.0          +  National Estimation  
30        Uruguay  2019  2019   338.0          +  National Estimation  
31        Uruguay  2020  2020   762.0          +  National Estimation  
45    Switzerland  2017  2017   147.0          +  National Estimation  
46    Switzerland  2018  2018   381.0          +  National Estimation  
...           ...   ...   ...     ...        ...                  ...  
2234       Israel  2017  2017  5994.0          +  National Estimation  
2235       Israel  2018  2018  6583.0          +  National Estimation  
2236       Israel  2019  2019  7161.0          +  National Estimation  
2237       Israel  2020  2020  7365.0          +  National Estimation  
2238       Israel  2021  2021  6891.0          +  National Estimation  

[732 rows x 9 columns]

# Gasto governamental em educação vs índice de crianças fora da escola

In [32]:
# Load the new dataset for government expenditure
government_expenditure = pd.read_csv('Government_exp_const_usd.csv')

# Load the dataset for children out of school again to ensure consistency
children_out_of_school = pd.read_csv('Children_out_of_school.csv')

# Display the first few rows of the new dataset to understand its structure
government_expenditure.head(), children_out_of_school.head()


(           NATMON_IND                                          Indicator  \
 0  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 1  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 2  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 3   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 4   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 
   LOCATION Country  TIME  Time        Value Flag Codes Flags  
 0      GRC  Greece  2017  2017    538.50359        NaN   NaN  
 1      GRC  Greece  2018  2018    557.72537        NaN   NaN  
 2      GRC  Greece  2019  2019    588.75234        NaN   NaN  
 3      IND   India  2017  2017  14367.64709        NaN   NaN  
 4      IND   India  2018  2018  15425.28270        NaN   NaN  ,
   NATMON_IND                                          Indicator LOCATION  \
 0  OFST_1_CP  Out-of-school children of primary school age, ...      MLT 

In [38]:
# Selecionar as colunas necessárias
children_out_of_school_filtered = children_out_of_school[['Country', 'Time', 'Value']].rename(columns={'Value': 'Children_Out_Of_School'})
government_expenditure_filtered = government_expenditure[['Country', 'Time', 'Value']].rename(columns={'Value': 'Government_Expenditure'})

# Combinar os datasets baseados em país e ano
merged_data_exp = pd.merge(children_out_of_school_filtered, government_expenditure_filtered, on=['Country', 'Time'])

# Remover linhas com valores NaN
cleaned_data_exp = merged_data_exp.dropna()

# Calcular a correlação para cada país
country_correlations_exp = cleaned_data_exp.groupby('Country').apply(
    lambda x: x['Children_Out_Of_School'].corr(x['Government_Expenditure'])
).reset_index().rename(columns={0: 'Correlation'})

# Criar o mapa com as correlações individuais dos países
fig_map_exp = px.choropleth(country_correlations_exp, 
                            locations="Country",
                            locationmode='country names',
                            color="Correlation",
                            hover_name="Country",
                            color_continuous_scale=px.colors.diverging.PiYG,
                            title="Correlation between Children Out of School and Government Expenditure by Country")

# Atualizar o layout para melhor legibilidade
fig_map_exp.update_layout(
    geo=dict(showframe=False, showcoastlines=True, projection_type='equirectangular')
)


In [ ]:
# Sort the correlations to get the top 10 highest and lowest correlations
top_10_highest_correlations = country_correlations_exp.sort_values(by='Correlation', ascending=False).head(10)
top_10_lowest_correlations = country_correlations_exp.sort_values(by='Correlation').head(10)

top_10_highest_correlations, top_10_lowest_correlations

Correlações Positivas: Alguns países como Camarões e Butão mostram correlações positivas relativamente altas. Isso sugere que, nesses países, maiores gastos governamentais em educação estão associados a um menor número de crianças fora da escola. Isso pode indicar que investimentos governamentais estão tendo um impacto positivo na inclusão escolar.

Correlações Negativas: Em outros países, como Moçambique e Paquistão, há correlações negativas significativas. Isso pode sugerir que, apesar dos gastos governamentais, outras barreiras (sociais, econômicas ou políticas) podem estar impedindo a redução do número de crianças fora da escola.

# Gasto governamental com educação vs população analfabeta

In [41]:
# Load the new dataset for government expenditure
government_expenditure = pd.read_csv('Government_exp_const_usd.csv')

# Load the dataset for children out of school again to ensure consistency
illetarate_population = pd.read_csv('Illetarate_population.csv')

# Display the first few rows of the new dataset to understand its structure
government_expenditure.head(), illetarate_population.head()


(           NATMON_IND                                          Indicator  \
 0  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 1  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 2  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 3   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 4   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 
   LOCATION Country  TIME  Time        Value Flag Codes Flags  
 0      GRC  Greece  2017  2017    538.50359        NaN   NaN  
 1      GRC  Greece  2018  2018    557.72537        NaN   NaN  
 2      GRC  Greece  2019  2019    588.75234        NaN   NaN  
 3      IND   India  2017  2017  14367.64709        NaN   NaN  
 4      IND   India  2018  2018  15425.28270        NaN   NaN  ,
              NATMON_IND                                          Indicator  \
 0  ILLPOP_AG15T24_URB_F  Illiterate youth population, 15-24 years, urba

In [43]:
# Selecionar as colunas necessárias
illetarate_population_filtered = illetarate_population[['Country', 'Time', 'Value']].rename(columns={'Value': 'Illetaracy'})
government_expenditure_filtered = government_expenditure[['Country', 'Time', 'Value']].rename(columns={'Value': 'Government_Expenditure'})

# Combinar os datasets baseados em país e ano
merged_data_exp = pd.merge(illetarate_population_filtered, government_expenditure_filtered, on=['Country', 'Time'])

# Remover linhas com valores NaN
cleaned_data_exp = merged_data_exp.dropna()

# Calcular a correlação para cada país
country_correlations_exp = cleaned_data_exp.groupby('Country').apply(
    lambda x: x['Illetaracy'].corr(x['Government_Expenditure'])
).reset_index().rename(columns={0: 'Correlation'})

# Criar o mapa com as correlações individuais dos países
fig_map_exp = px.choropleth(country_correlations_exp, 
                            locations="Country",
                            locationmode='country names',
                            color="Correlation",
                            hover_name="Country",
                            color_continuous_scale=px.colors.diverging.PiYG,
                            title="Correlation between levels of illetaracy and Government Expenditure by Country")

# Atualizar o layout para melhor legibilidade
fig_map_exp.update_layout(
    geo=dict(showframe=False, showcoastlines=True, projection_type='equirectangular')
)


In [44]:
# Sort the correlations to get the top 10 highest and lowest correlations
top_10_highest_correlations = country_correlations_exp.sort_values(by='Correlation', ascending=False).head(10)
top_10_lowest_correlations = country_correlations_exp.sort_values(by='Correlation').head(10)

top_10_highest_correlations, top_10_lowest_correlations

(        Country  Correlation
 37         Oman     0.073263
 38     Pakistan     0.016172
 31         Mali     0.009204
 51    Sri Lanka     0.004894
 22    Guatemala     0.001872
 18      Ecuador     0.001190
 20  El Salvador     0.001089
 34       Mexico     0.000466
 41         Peru     0.000437
 8        Brazil     0.000081,
                  Country  Correlation
 21               Georgia    -0.258769
 56            Uzbekistan    -0.049874
 19                 Egypt    -0.024160
 44                Rwanda    -0.015815
 49          South Africa    -0.006111
 54  United Arab Emirates    -0.005134
 14              Colombia    -0.003815
 2                Armenia    -0.003194
 52               Türkiye    -0.002611
 48             Singapore    -0.002136)

Correlações Positivas Fracas:
A maioria das correlações positivas entre o gasto governamental e a população analfabeta são extremamente fracas, próximas de zero. Isso sugere que, nesses países, o aumento do gasto governamental não está significativamente associado a uma redução da população analfabeta.
Exemplos como Omã (0.0733) mostram uma correlação positiva ligeiramente maior, mas ainda muito fraca, indicando que há uma relação muito limitada entre os gastos e a taxa de alfabetização.

Correlações Negativas:
Alguns países, como Geórgia (-0.2588) e Uzbequistão (-0.0499), apresentam correlações negativas mais significativas. Isso sugere que, nesses países, um maior gasto governamental está associado a uma diminuição na população analfabeta, indicando que os investimentos governamentais podem estar contribuindo para melhorar a alfabetização.
No entanto, essas correlações negativas também não são muito fortes, indicando que, embora possa haver algum efeito, ele não é dominante.

# Gasto governamental em educação vs taxa de crianças que completam a educação primária

In [45]:
# Load the new dataset for government expenditure
government_expenditure = pd.read_csv('Government_exp_const_usd.csv')

# Load the dataset for children out of school again to ensure consistency
survival_rate_ed = pd.read_csv('Survival_rate_primary_ed.csv')

# Display the first few rows of the new dataset to understand its structure
government_expenditure.head(), survival_rate_ed.head()


(           NATMON_IND                                          Indicator  \
 0  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 1  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 2  X_USCONST_02_FSGOV  Government expenditure on pre-primary educatio...   
 3   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 4   X_USCONST_2_FSGOV  Government expenditure on lower secondary educ...   
 
   LOCATION Country  TIME  Time        Value Flag Codes Flags  
 0      GRC  Greece  2017  2017    538.50359        NaN   NaN  
 1      GRC  Greece  2018  2018    557.72537        NaN   NaN  
 2      GRC  Greece  2019  2019    588.75234        NaN   NaN  
 3      IND   India  2017  2017  14367.64709        NaN   NaN  
 4      IND   India  2018  2018  15425.28270        NaN   NaN  ,
         NATMON_IND                                          Indicator  \
 0  SR_1_GLAST_F_CP  Survival rate to the last grade of primary edu...   
 1 

In [49]:
# Selecionar as colunas necessárias
survival_rate_ed_filtered = survival_rate_ed[['Country', 'Time', 'Value']].rename(columns={'Value': 'Survival_Rate_Ed'})
government_expenditure_filtered = government_expenditure[['Country', 'Time', 'Value']].rename(columns={'Value': 'Government_Expenditure'})

# Combinar os datasets baseados em país e ano
merged_data_exp = pd.merge(survival_rate_ed_filtered, government_expenditure_filtered, on=['Country', 'Time'])

# Remover linhas com valores NaN
cleaned_data_exp = merged_data_exp.dropna()

# Calcular a correlação para cada país
country_correlations_exp = cleaned_data_exp.groupby('Country').apply(
    lambda x: x['Survival_Rate_Ed'].corr(x['Government_Expenditure'])
).reset_index().rename(columns={0: 'Correlation'})

# Criar o mapa com as correlações individuais dos países
fig_map_exp = px.choropleth(country_correlations_exp, 
                            locations="Country",
                            locationmode='country names',
                            color="Correlation",
                            hover_name="Country",
                            color_continuous_scale=px.colors.diverging.PiYG,
                            title="Correlação entre a sobrevivência no ensino primário vs gastos governamentais")

# Atualizar o layout para melhor legibilidade
fig_map_exp.update_layout(
    geo=dict(showframe=False, showcoastlines=True, projection_type='equirectangular')
)


In [48]:
# Sort the correlations to get the top 10 highest and lowest correlations
top_10_highest_correlations = country_correlations_exp.sort_values(by='Correlation', ascending=False).head(10)
top_10_lowest_correlations = country_correlations_exp.sort_values(by='Correlation').head(10)

top_10_highest_correlations, top_10_lowest_correlations

(                         Country  Correlation
 20                      Cameroon     0.311621
 86                  Sierra Leone     0.089773
 99   United Republic of Tanzania     0.057574
 66                    Mozambique     0.033843
 35                         Egypt     0.030731
 103                      Vanuatu     0.014649
 5                     Azerbaijan     0.014297
 102                   Uzbekistan     0.002554
 30                 Côte d'Ivoire     0.002342
 85                        Serbia     0.001946,
                                             Country  Correlation
 16                                     Burkina Faso    -0.133351
 76                                            Qatar    -0.069870
 81                                           Rwanda    -0.038538
 104                                        Zimbabwe    -0.022367
 33                               Dominican Republic    -0.017762
 17                                          Burundi    -0.012057
 23   China, Hong Ko

Correlações Positivas: Camarões (0.3116) mostra uma correlação positiva relativamente alta, indicando que um aumento nos gastos governamentais em educação está associado a uma maior taxa de crianças completando o ensino primário. Este pode ser um exemplo de como investimentos adequados em educação podem ter um impacto positivo na conclusão do ensino primário.
Outros países como Serra Leoa e Tanzânia também mostram correlações positivas, embora mais fracas, sugerindo algum nível de associação entre os gastos e a taxa de conclusão do ensino primário.

Correlações Negativas: Burkina Faso (-0.1334) mostra a correlação negativa mais significativa, sugerindo que, neste caso, maiores gastos governamentais em educação podem estar associados a uma menor taxa de conclusão do ensino primário. Isso pode indicar problemas de eficiência no uso dos recursos ou outros fatores contextuais que afetam negativamente a taxa de conclusão.
Outros países como Catar e Ruanda também apresentam correlações negativas, embora mais fracas, indicando que o aumento dos gastos governamentais não está necessariamente resultando em melhores taxas de conclusão do ensino primário.